src/coreclr/src/tools/r2rdump/R2RDiff.cs

```diff
@@ -75,46 +69,38 @@ public void Run()
                 {
                     int leftIndex = _leftDumper.Reader.ManifestReferenceAssemblies[assemblyName];
                     int rightIndex = _rightDumper.Reader.ManifestReferenceAssemblies[assemblyName];
-
-                    if (leftIndex < 0 ||  !_leftDumper.Reader.ReadyToRunAssemblyHeaders[leftIndex].Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out leftSection))
-                    {
-                        leftSection = new ReadyToRunSection();
-                    }
-                    if (rightIndex < 0 || !_rightDumper.Reader.ReadyToRunAssemblyHeaders[rightIndex].Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out rightSection))
-                    {
-                        rightSection = new ReadyToRunSection();
-                    }
-                    _writer.WriteLine($@"{assemblyName}: component method diff");
-                    DiffMethodsForModule(leftSection, rightSection);
+                    DiffMethodsForModule(leftIndex, rightIndex);
                 }
             }
         }
 
-        private bool TryGetMethods(ReadyToRunReader reader, ReadyToRunSection section, out IReadOnlyList<ReadyToRunMethod> methods)
+        private IEnumerable<ReadyToRunMethod> TryGetMethods(ReadyToRunReader reader, int moduleIndex)
         {
-            int assemblyIndex = reader.GetAssemblyIndex(section);
-            if (assemblyIndex == -1)
+            List<ReadyToRunMethod> methods = new List<ReadyToRunMethod>();
+            switch (moduleIndex)
             {
-                methods = null;
-                return false;
-            }
-            else
-            {
-                methods = reader.ReadyToRunAssemblies[assemblyIndex].Methods;
-                return true;
+                case InvalidModule:
+                    break;
+
+                case AllModules:
+                    methods.AddRange(reader.InstanceMethods.Select(im => im.Method));
+                    foreach (ReadyToRunAssembly assembly in reader.ReadyToRunAssemblies)
+                    {
+                        methods.AddRange(assembly.Methods);
+                    }
+                    break;
+
+                default:
+                    methods.AddRange(reader.ReadyToRunAssemblies[moduleIndex].Methods);
+                    break;
             }
+            return methods;
         }
 
-        private void DiffMethodsForModule(ReadyToRunSection leftSection, ReadyToRunSection rightSection)
+        private void DiffMethodsForModule(int leftIndex, int rightIndex)
```

Comment:
A micro nit - would you mind naming the arguments `leftModuleIndex` and `rightModuleIndex` instead for better clarity of what kind of index they represent?

Reply:
Good point, renamed in 2nd commit (in three different places in the file).